src/consumer.rs

```diff
@@ -372,17 +411,18 @@ enum InnerConsumer<T: DeserializeMessage, Exe: Executor> {
 
 type MessageIdDataReceiver = mpsc::Receiver<Result<(proto::MessageIdData, Payload), Error>>;
 
+// this is entirely public for use in reader.rs
 pub(crate) struct TopicConsumer<T: DeserializeMessage, Exe: Executor> {
-    consumer_id: u64,
-    config: ConsumerConfig,
-    topic: String,
-    messages: Pin<Box<MessageIdDataReceiver>>,
-    engine_tx: mpsc::UnboundedSender<EngineMessage<Exe>>,
+    pub(crate) consumer_id: u64,
```

Comment:
do we need all of those to be pub(crate)?

Reply:
as a matter of fact, no, we don't. I ran `cargo check`ed all of them, only `consumer_id`, `config`, and `dead_letter_policy` are needed elsewhere.